articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +219,83 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
+However, aliasing is easily douable with it.
+That is, having an original topic such as `cmd_vel` in a namespace `robot1`, we can easily duplicate that topic as `robot2/cmd_vel`.
+
+#### Hierarchy with DDS Partitions
+
+Given that DDS Partitions are not hierarchical, we have to setup the hierarchy on top.
+For this, we use the forward slash `/` explicitely in each partition field.
+If we want to replicate a hierarchy of `warehouse`, `robot1`, `camera_left`, and finally the basename `image_raw`, the first entry of the partitions would look like:
+
+```
+warehouse/robot1/camera_left/image_raw
+```
+
+#### Remapping with DDS Partitions
+
+We have to differentiate remapping from aliasing at this point.
+As described before, aliasing implies a duplication a topic.
+Remapping, in contrast, alters an existing topic and thus remaps one topic into another.
+Given the nature of DDS Partitions, aliasing is natively supported by adding a second entry in the partition field.
+Remapping means, we have to change one existing partition field.
+However, as the completely hierarchy is illustrated within the first index of the partitions, the act of remapping is a simple string modification.
+If we want to remap a camera image from `camera1` to `camera2`, we have to modify the string in the respective field.
+Publishers or Subscribers are uneffected by this change and don't have to be shutdown and restarted.
+The way of modifying these strings, whether with complete string replacement or regex modifiers, exceeds the scope of this article.
+
+## Compare and Contrast with ROS 1
+
+In order to support a mapping to the more restrictive DDS topic name rules, these rules are in some ways more constraining than the rules for ROS 1.
+Other changes have been proposed for convenience or to remove a point of confusion that existed in ROS 1.
+In ROS 2, topic and service names differ from ROS 1 in that they:
+
+- must separate the tilde (`~`) from the rest of the name with a forward slash (`/`)
+
+  - This is done to avoid inconsistency with how `~foo` works in filesystem paths versus when used in a ROS name.
+
+- may contain substitutions which are delimited with balanced curly braces (`{}`)
+
+  - This is a more generic extension of the idea behind the tilde (`~`).
+
+- have length limits
+
+  - This is driven by the topic name length limit of DDS.
+
+- may be indicated as "hidden" by using a leading underscore (`_`) in one of the namespaces
+
+  - This is used to hide common but infrequently introspected topics and services.
+
+## Concerns/Alternatives
+
+This section lists concerns about the proposed design and alternatives that were considered.
 
 ### ROS Specific Name Prefix
 
+Initially, the idea of implementing ROS topics and namespaces based on specific name prefixes was chosen.
+This is not any longer the favorite solution as DDS partitions have proven to be more flexible and accurate since they are part of the protocol.
```

Comment:
I may have misunderstood something here. Why I removed that section was because of not prefixing topics with `rt` and `rx` etc. 
But I can leave it in if you wish.